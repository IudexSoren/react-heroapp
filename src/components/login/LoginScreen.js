import { useContext } from "react"
import { AuthContext } from "../../auth/AuthContext"
import { types } from "../../types/types";


const LoginScreen = ({ history }) => {

  const { dispatch } = useContext(AuthContext);

  const handleLogin = () => {
    const action = {
      type: types.login,
      payload: {
        name: 'Aar√≥n'
      }
    }
    dispatch(action);
    const lastPath = localStorage.getItem('lastPath') || '/';
    history.replace(lastPath);
  }

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr/>
      <button
        className="btn btn-outline-dark"
        onClick={ handleLogin }
      >
        Log In
      </button>
    </div>
  )
}

export default LoginScreen;
