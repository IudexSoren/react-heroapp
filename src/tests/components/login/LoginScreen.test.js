import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/AuthContext";
import LoginScreen from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";


describe('Pruebas de LoginScreen', () => {

  const history = {
    replace: jest.fn()
  }

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  }

  const wrapper = mount(
    <AuthContext.Provider value={ contextValue }>
      <MemoryRouter>
        <LoginScreen history={ history } />
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('Debe renderizar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe ejecutar el dispatch y la navegación', () => {
    wrapper.find('button').simulate('click');

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'Aarón'
      }
    });
    expect(history.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPath', '/dc');
    wrapper.find('button').simulate('click');

    expect(history.replace).toHaveBeenCalledWith('/dc');
  });

});
