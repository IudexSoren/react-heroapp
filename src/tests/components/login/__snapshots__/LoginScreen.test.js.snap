// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de LoginScreen Debe renderizar correctamente el componente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-dark"
    onClick={[Function]}
  >
    Log In
  </button>
</div>
`;
